### PR TITLE
Allow adding all contacts to distribution lists

### LINKED ISSUE #836: Distribution Lists: Cannot add members without GROUP_CHAT capability
## Bug Description

I cannot add members without the group chat feature flag to distribution lists, even though that should be allowed.

## Steps to Reproduce (for bugs)

- Adding ECHOECHO to groups should not work
- Adding ECHOECHO to distribution lists should work

### PR DESCRIPTION
**Pull request by dbrgn**
Fixes #836.

### CODE REVIEW

File: src/controller_model/distributionList.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export class DistributionListControllerModel implements threema.ControllerModel<
     public name: string;
     public subject: string;
     public isLoading = false;
+    public readonly requiredMemberFeatureMask = null; // No constraints
```
Comment (lgrahl) [inline src/controller_model/distributionList.ts]: Do you mean `any` by that? But it clearly isn't `any`.

File: src/controller_model/group.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export class GroupControllerModel implements threema.ControllerModel<threema.Gro
     public name: string;
     public access: threema.GroupReceiverAccess;
     public subject: string;
+    public readonly requiredMemberFeatureMask = threema.ContactReceiverFeature.GROUP_CHAT;
```
Comment (lgrahl) [inline src/controller_model/group.ts]: Type missing.

File: src/controller_model/distributionList.ts (hunk 1 above)
Comment (dbrgn) [reply src/controller_model/distributionList.ts]: I can't follow. What do you mean?

**Comment by dbrgn** [comment]
@lgrahl type added to an interface.

File: src/helpers.ts [hunk 3]
```diff
@@ -290,6 +290,9 @@ export function hasFeature(contactReceiver: threema.ContactReceiver,
             log.warn(`Contact receiver with id ${contactReceiver.id} has featureMask 0`);
             return false;
         }
+        if (feature === threema.ContactReceiverFeature.NONE) {
+            return true;
+        }
```
Comment (dbrgn) [inline src/helpers.ts]: :confused:

**Comment by dbrgn** [comment]
@lgrahl refactoring done.

File: src/helpers.ts (hunk 3 above)
Comment (lgrahl) [reply src/helpers.ts]: Slightly awkward but not any more awkward than the `undefined` check above. :smirk_cat:

File: src/controller_model/distributionList.ts [hunk 4]
```diff
@@ -22,18 +22,28 @@ import {WebClientService} from '../services/webclient';
...
     private $mdDialog: ng.material.IDialogService;
+
+    // Custom services
     private readonly log: Logger;
-    public members: string[];
-    public name: string;
+    private readonly webClientService: WebClientService;
+
+    // Fields required by interface
+    public readonly receiverType = 'contact';
```
Comment (dbrgn) [inline src/controller_model/distributionList.ts]: Oh nice. How did this not break something when testing it?!

One of the nice side effects of trying to add more structure, it increases the surface for bugs.

Comment (dbrgn) [reply src/controller_model/distributionList.ts]: No failure because unused (messenger uses this for mentions, which are only allowed in groups).

**Pull request merged**